src/roiextractors/multiimagingextractor.py

```diff
@@ -97,16 +106,53 @@ def get_frames(self, frame_idxs: ArrayType, channel: Optional[int] = 0) -> Numpy
     def get_video(
         self, start_frame: Optional[int] = None, end_frame: Optional[int] = None, channel: int = 0
     ) -> np.ndarray:
-        # To-do: implement this without reference to get_frames
+        if channel != 0:
+            raise NotImplementedError(
+                f"MultiImagingExtractors for multiple channels have not yet been implemented! (Received '{channel}'."
+            )
+
         start = start_frame if start_frame is not None else 0
         stop = end_frame if end_frame is not None else self.get_num_frames()
-        frame_idxs = range(start, stop)
-        return self.get_frames(frame_idxs=frame_idxs, channel=channel)
+        total_range = list(range(start, stop))
+        extractors_spanned = np.unique(np.searchsorted(self._end_frames, total_range, side="right"))
```

Comment:
Clever!

Reply:
Could be more clever. As is, still generated an entire vector of length `stop - start`, so for a really long imaging extractor could get more expensive than necessary

Reply:
Anyway, fixed that. Only uses the two bounds now